metabase/metabase
# Display full-dates for trend charts backed by native queries

### Problem

For native queries, the chosen display date, for an aggregated value across range a range of dates, is up to the creator. This is unlike the query builder which chooses the display date (across an aggregated range of dates) to be at the start of the range. Trend charts expects a date for a given `dateUnit` to be at the start of the range as well.
- Examples:
  - Aggregating by month
    - Query Builder: Start of the month (Jan. 1, Feb. 1, Mar. 1, ...)
    - Native Query: Whatever the creator decides (Jan 1 or Jan 31, Feb 1 or Feb 28, ...)
  - Aggregating by year
    - Query Builder: Start of the year (Jan 1 2022, Jan 1 2023, ...)
    - Native Query: Whatever the creator decides (Jan 1 or Dec 31 2022, Jan 1 or Dec 31 2023, ...)

---
### Desired Behavior/Acceptance Criteria
**Solution Behaviors**
- For native queries
   ![image](https://github.com/metabase/metabase/assets/22608765/913907a7-f03b-4321-8f14-466879be7893)
  - [ ] always display the full-date (disregard the `dateUnit` only for formatting)
  - [ ] keep the inferred `dateUnit` behavior to still allow `periods ago` comparisons
- [ ] All other behaviors should be unaltered

**Validation Tests**
- [ ] (`compute.unit.spec.js`) Native queries always display the full-date regardless of the inferred `dateUnit`

---
### How to Manually Verify the Solution
1. Follow the steps listed in the issue
   - https://github.com/metabase/metabase/issues/38122
2. See that the full-dates are displayed instead of the condensed `dateUnit` versions
3. See that there are still options for `previous period` and `periods ago` comparisons
4. Create a regular (non-native) query and see that trend charts displays the condensed versions of the dates

---
**Context**
- Will close after backport: https://github.com/metabase/metabase/issues/38122
